### PR TITLE
fix: login and register page with validation errors

### PR DESCRIPTION
**Pull request by Pushpendra33**
### Related Issue 
issue number #735 

#### Changes made 👷🏻‍♂️
Improved login and register form with global compoents and validation (yup).
With this update, users will be able to access the application using their credentials, providing an additional layer of error throwing on the login and registration process.
With react-hook-form user have to pass all the validations to perform sign up or login.

### Packages installed by my side
1. @hookform/resolvers
2. react-hook-form
3. yup

#### Screenshots  📸
Screenshot from 2023-06-05 12-20-07

### CODE REVIEW

**Comment by tamalCodes** [comment]
Hi @Pushpendra33 , that's a lot of files to go through  and review properly - I will take it slowly, thanks for working so hard on it !

**Comment by Pushpendra33** [comment]
> Hi @Pushpendra33 , that's a lot of files to go through and review properly - I will take it slowly, thanks for working so hard on it !

Thanks for the appreciation @tamalCodes , take your time :+1: .

File: src/pages/clubs/ClubsRegister.jsx
Comment (tamalCodes) [inline src/pages/clubs/ClubsRegister.jsx:1]: For Club's register, we are not adding validations ? I checked [here](https://milan-phhl7dm48-iamtamal.vercel.app/clubs/register) too, but did not find anything.

**Review by tamalCodes — CHANGES_REQUESTED** [review]
We also need some changes based on devices. 
For example to the left are the changes you made, and to the right is the original one.

image

**Comment by Pushpendra33** [comment]
All the changes requested from your side are completed.
Please check and go through it

**Comment by tamalCodes** [comment]
I am closing this PR as it was requested by @Pushpendra33. We will continue with #850

**Pull request closed without merging**